src/blueapi/cli/cli.py

```diff
@@ -19,24 +20,30 @@
 @click.pass_context
 def main(ctx, config: Optional[Path]) -> None:
     # if no command is supplied, run with the options passed
+
     config_loader = ConfigLoader(ApplicationConfig)
     if config is not None:
         config_loader.use_values_from_yaml(config)
 
     ctx.ensure_object(dict)
-    ctx.obj["config"] = config_loader.load()
+    ctx.obj["config_loader"] = config_loader
```

Comment:
Feel like this was better the old way. Now you have to call `load()` in two places instead of one

Reply:
Yes, the reason I did this, is because pydantic basemodels are not hashable. This meant I couldn't call `get_handler` with the ApplicationConfig because it has a lru_cache which by default puts it into a dictionary... so I initially went around this by just passing around the ConfigLoader instead.

However... I have since got rid of the lru_cache and decided the easiest way to deal with this is via global variables. In which case I don't need it. I will try passing around just the ApplicationConfig because I agree it's a lot neater.

Reply:
edit: yep, have changed it back and it works perfectly fine.